source/develop/release-management/features/gluster/gdeploy-cockpit-integration.html.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Gdeploy Cockpit Integration
+category: feature
+authors: rnachimu
+wiki_category: Feature|Gdeploy-Cockpit integration
+wiki_title: Feature/GdeployCockpitIntegration
+wiki_revision_count: 0
+wiki_last_updated: 2016-09-19
+feature_name: Gdeploy Integration with Cockpit
+feature_modules: cockpit plugin for ovirt
+feature_status: WIP, 4.1 proposed feature
+---
+
+#Gdeploy integration with Cockpit-oVirt Plugin
+
+## Summary
+With the introduction of Cockpit-oVirt Plugin in [oVirt Node NG](http://www.ovirt.org/develop/projects/node/4.0), it is possible to deploy hosted-engine using cockpit. But in a Hyper-Converged environment with Gluster, we still need to configure Glusterfs storage manually before deploying hosted engine.
+Gdeploy is an ansible based tool to deploy GlusterFS Servers. It can be used to deploy Glusterfs over multiple nodes.
+
+This feature describes a mechanism to integrate gdeploy and Cockpit-oVirt Plugin to facilitate the easy deployment of Hyper-Converged Ovirt-Gluster environment.
+
+### Cockpit-oVirt Plugin
+The [Cockpit-oVirt plugin](https://www.ovirt.org/develop/release-management/features/cockpit) is a [Cockpit](http://cockpit-project.org) plugin which provides VM-centric monitoring and management of a host machine while taking advantage of already built-in admin functionality in the Cockpit.
+This also has a provision to deploy oVirt Hosted-Engine through Cockpit UI.
+
+### Gdeploy
+[gdeploy](https://github.com/gluster/gdeploy) is a tool to set-up and deploy GlusterFS using ansible over multiple hosts. gdeploy is written to be modular, it can be used to deploy any software depending on how the configuration file is written.
+gdeploy can be used to set-up bricks for GlusterFS, create a GlusterFS volume and mount it on one or more clients from an ansible installed machine. The framework reads a configuration file and applies on the hosts listed in the configuration file.
+
+### Owner
+* Name: Ramesh Nachimuthu
+* Email: rnachimu@redhat.com
+
+### Current Status
+* Status: Design
+* Last updated date: 20th Sep 2016
+
+### Benefit to oVirt
+
+This feature will greatly improve the user experience of deploying oVirt and Gluster hyperconverged solution using oVirt Node NG. oVirt users will be benefited in following ways.
+
+* Easy deployment using Cockpit UI
+* Well integrated deployment of HCI solution
+* Consistent Gluster deployment using gdeploy
+
+### Dependencies
+gdeploy package should be installed and running. gdeploy requires password less ssh from the gdeploy host to other hosts where we want to deploy gluster. This needs to be done manually before starting deployment flow.
+
+**TBD:** explore a way to configure password less ssh using Cockpit itself.
+
```

Comment:
Since the deployment is intended to work on newly added host without the need forf excessive pre-configuration, I think the easiest way to achieve the pwd-less ssh is just to scp public keys, so the password needs to be entered just once during this initialization process.

I don't think there's some explicit support for this task in cockpit so far.